ios/RNSentryRNSScreen.m

```diff
@@ -0,0 +1,26 @@
+#import <Sentry/SentryFramesTracker.h>
+#import <Sentry/SentryDependencyContainer.h>
+#import <Sentry/SentrySwizzle.h>
+
+#import "RNSentryRNSScreen.h"
+#import "RNSentryDependencyContainer.h"
+
+@implementation RNSentryRNSScreen
+
++ (void)swizzleViewDidAppear {
+  Class rnsscreenclass = NSClassFromString(@"RNSScreen");
+  if (rnsscreenclass == nil)
+  {
+    return;
+  }
+
+  SEL selector = NSSelectorFromString(@"viewDidAppear:");
+  SentrySwizzleInstanceMethod(rnsscreenclass, selector, SentrySWReturnType(void),
+      SentrySWArguments(BOOL animated), SentrySWReplacement({
+          [[[RNSentryDependencyContainer sharedInstance] framesTrackerListener] startListening];
+          SentrySWCallOriginal(animated);
+      }),
+      SentrySwizzleModeOncePerClass, (void *)selector);
+}
```

Comment:
h: Now that Im reading the full code, I dont think you should start listening when the viewDidAppear is called,
you can start listening at the beginning of the app. When viewDidAppear is called you just report a for the ongoing UI transaction.

Reply:
So instead of adding a listener in `viewDidAppear` I would set some flag and then I would wait for the next frame?

Won't that cause https://github.com/getsentry/sentry-react-native/blob/89818402b1846d4170a03414a4b495d63433b8fc/ios/RNSentryFramesTrackerListener.m#L16 to be executed for every frame in the app?

Or did I misunderstand?

Reply:
I see. you remove the listener from frame track every time you emit and event. I haven't notice it before.
Well it's almost like using a flag, yeah, I dont think you need to change this.